modules/asciival/BESAsciiTransmit.cc

```diff
@@ -98,6 +99,10 @@ void BESAsciiTransmit::send_basic_ascii(BESResponseObject *obj, BESDataHandlerIn
         // Send data values as CSV/ASCII
         unique_ptr<DDS> ascii_dds(datadds_to_ascii_datadds(loaded_dds));  // unique_ptr<> jhrg 9/6/16
 
+        // Verify the request hasn't exceeded bes_timeout, and disable timeout if allowed.
+        RequestServiceTimer::TheTimer()->throw_if_timeout_expired(prolog + " Ready to start streaming", __FILE__, __LINE__);
```

Comment:
This is the hard part where to put the test(s). I wonder (but I don't know) if it would be better to test inside get_data_values_as_ascii(...); I don't know - maybe that's overkill. Here's the bulk of that function:

    DDS::Vars_iter i = dds->var_begin();
    while (i != dds->var_end()) {
        if ((*i)->send_p()) {
            dynamic_cast<AsciiOutput &>(**i).print_ascii(strm);
            strm << "\n";
        }
        ++i;
    }

If we tested the timer at the end of the loop, then we'd be testing once some data are sent. But then the time would include the time to send the data...

I think your solution is probably what we want, but we might need to revisit this later.